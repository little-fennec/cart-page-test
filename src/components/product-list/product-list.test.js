import React from 'react';
import ProductList from './index';
import Enzyme, {mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

Enzyme.configure({ adapter: new Adapter() });

describe('Screen <ProductList/>', () => {

    const initialState = {
        loading: false,
        cartItems: [],
        items:[]
    };

    const store = mockStore(initialState);

    it('renders correctly ', () => {
        const wrapper = mount(
            <Provider store={store}>
                    <ProductList/>
            </Provider>
        );
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
