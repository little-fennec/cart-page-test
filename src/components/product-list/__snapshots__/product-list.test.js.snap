// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen <ProductList/> renders correctly  1`] = `
<div
  className="product-list"
/>
`;
